### DISCUSSION TITLE
error: no such command: `+solana`

### DISCUSSION THREAD

**Discussion by virjilakrum** [otter-sec/anchor#2831] [Q&A] [answered]
<img width="2056" alt="Ekran Resmi 2024-03-04 12 44 48" src="https://github.com/coral-xyz/anchor/assets/158029357/f96913df-bad5-4366-adae-9eb2a1c6d93f">

▸ The steps I follow

```
anchor init tokenswap_contract --javascript

curl --proto '=https' --tlsv1.2 -sSf https://sh.rustup.rs | sh

rustc --version

cargo install --git https://github.com/coral-xyz/anchor anchor-cli --locked

anchor --version

sh -c "$(curl -sSfL https://release.solana.com/stable/install)"

solana --version

anchor init tokenswap_contract --javascript

cd tokenswap_contract

anchor build
```

▸ It explodes here because there is no configurator called "+solana" in the rustup toolchain

🌐 Anyone help me about this ?

**Comment by acheroncrypto** [comment]
It's likely that your Rust installation is causing this problem. I've seen people run into this problem with `brew` installation on Apple devices, so I'd run:

```
brew uninstall rust
```

and install from the official Rust website: https://www.rust-lang.org/tools/install

**Reply by C-o-d-e-C-o-w-b-o-y** [reply, marked as answer]
> It's likely that your Rust installation is causing this problem. I've seen people run into this problem with `brew` installation on Apple devices, so I'd run:
> 
> ```
> brew uninstall rust
> ```
> 
> and install from the official Rust website: https://www.rust-lang.org/tools/install

This fixed it for me after many hours of frustration!!!

https://x.com/Code_Cowboy_/status/1852941956474622206

**Reply by williamhgough** [reply]
This fixed it for me! Thank you!
